controller/src/main/java/io/pravega/controller/server/rpc/grpc/v1/ControllerServiceImpl.java

```diff
@@ -477,19 +485,38 @@ public void getDelegationToken(StreamInfo request, StreamObserver<DelegationToke
         authenticateExecuteAndProcessResults(() -> this.grpcAuthHelper.checkAuthorizationAndCreateToken(
                 AuthResourceRepresentation.ofStreamInScope(request.getScope(), request.getStream()),
                 AuthHandler.Permissions.READ_UPDATE),
-                delegationToken -> CompletableFuture.completedFuture(Controller.DelegationToken
-                        .newBuilder()
-                        .setDelegationToken(delegationToken)
-                        .build()),
+                delegationToken -> {
+                    logIfEmpty(delegationToken, "getDelegationToken", request.getScope(), request.getStream());
+                    return CompletableFuture.completedFuture(DelegationToken
+                            .newBuilder()
+                            .setDelegationToken(delegationToken)
+                            .build());
+                },
                 responseObserver);
     }
 
+    private void logIfEmpty(String delegationToken, String requestName, String scopeName,
+                            String streamName) {
+        if (isAuthEnabled()) {
+            String message = String.format("Delegation token for request [%s] with scope [%s] and stream [%s], is: [%s]",
```

Comment:
This string should only be constructed if we are sure we are going to output it. Otherwise, we will be constructing these strings unnecessarily. Please reorganize this logic across these two methods to log if empty to avoid constructing these strings unnecessarily.

Reply:
Sure, that makes sense. I've modified the code and pushed the changes.